nikola/plugins/task/archive.py

```diff
@@ -24,231 +24,146 @@
-        posts = sorted(posts, key=lambda a: a.date)
-        posts.reverse()
-        if kw['archives_are_indexes']:
-            def page_link(i, displayed_i, num_pages, force_addition, extension=None):
-                feed = "_atom" if extension == ".atom" else ""
-                return adjust_name_for_index_link(self.site.link("archive" + feed, name, lang), i, displayed_i,
-                                                  lang, self.site, force_addition, extension)
-
-            def page_path(i, displayed_i, num_pages, force_addition, extension=None):
-                feed = "_atom" if extension == ".atom" else ""
-                return adjust_name_for_index_path(self.site.path("archive" + feed, name, lang), i, displayed_i,
-                                                  lang, self.site, force_addition, extension)
-
-            uptodate = []
-            if deps_translatable is not None:
-                uptodate += [config_changed(deps_translatable, 'nikola.plugins.task.archive')]
-            context = {"archive_name": name,
-                       "is_feed_stale": kw["is_feed_stale"],
-                       "pagekind": ["index", "archive_page"]}
-            yield self.site.generic_index_renderer(
-                lang,
-                posts,
-                title,
-                "archiveindex.tmpl",
-                context,
-                kw,
-                str(self.name),
-                page_link,
-                page_path,
-                uptodate)
+            # Fallback
+            return '/'.join(classification)
+
+    def get_path(self, classification, lang, type='page'):
+        """A path handler for the given classification."""
+        components = [self.site.config['ARCHIVE_PATH']]
+        if classification:
+            components.extend(classification)
+            add_index = 'always'
         else:
-            yield self._prepare_task(kw, name, lang, posts, None, "list_post.tmpl", title, deps_translatable)
+            components.append(os.path.splitext(self.site.config['ARCHIVE_FILENAME'])[0])
+            add_index = 'never'
+        return [_f for _f in components if _f], add_index
+
+    def extract_hierarchy(self, classification):
+        """Given a classification, return a list of parts in the hierarchy."""
+        return classification.split('/') if classification else []
 
-    def gen_tasks(self):
-        """Generate archive tasks."""
+    def recombine_classification_from_hierarchy(self, hierarchy):
+        """Given a list of parts in the hierarchy, return the classification string."""
+        return '/'.join(hierarchy)
+
+    def provide_context_and_uptodate(self, classification, lang):
+        """Provide data for the context and the uptodate list for the list of the given classifiation."""
+        hierarchy = self.extract_hierarchy(classification)
         kw = {
             "messages": self.site.MESSAGES,
-            "translations": self.site.config['TRANSLATIONS'],
```

Comment:
`output_folder`, `archives_are_indexes`, `create_monthly_archives`, `create_single_archive`, `show_untranslated_posts`, `create_full_archives`, `create_daily_archive` and `generate_atom` are handled indirectly by which tasks are created. `messages` and `output_folder` are handled by `RenderTaxonomies._generate_classification_page`. The options `pretty_urls`, `strip_indexes` and `index_file` are missing, but they should also be better handled by `RenderTaxonomies._generate_classification_page`.